### PR TITLE
Enable building on macOS and Linux

### PR DESCRIPTION
**Pull request by bricelam**

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -5,6 +5,7 @@
     <SemanticVersioningV1>True</SemanticVersioningV1>
     <IncludeSourceRevisionInInformationalVersion>False</IncludeSourceRevisionInInformationalVersion>
     <UsingToolNetFrameworkReferenceAssemblies>True</UsingToolNetFrameworkReferenceAssemblies>
+    <MicrosoftNetFrameworkReferenceAssembliesVersion>1.0.0-preview.1</MicrosoftNetFrameworkReferenceAssembliesVersion>
```
Comment (bricelam) [inline eng/Versions.props]: Looks like this version still doesn't have the fix for dsplaisted/ReferenceAssemblyPackages#2

cc @dsplaisted

Comment (bricelam) [reply eng/Versions.props]: cc @NikolaMilosavljevic

Comment (dsplaisted) [reply eng/Versions.props]: We have this tracked at https://github.com/dotnet/core-sdk/issues/1912.  It should be fixed soon.

Comment (bricelam) [reply eng/Versions.props]: Thanks!

Comment (dsplaisted) [reply eng/Versions.props]: We just published an update that should fix this: https://www.nuget.org/packages/Microsoft.NETFramework.ReferenceAssemblies/1.0.0-preview.2

**Comment by bricelam** [comment]
Everything appears to be working as expected now. Development on macOS and Linux is unblocked.

We're not going to enable macOS and Ubuntu on the CI since there are still 4K tests failing with "LocalDB is not supported on this platform." If the community is interested in skipping these tests, or getting the base connection string from an environment variable, we'd happily accept a PR.

**Pull request merged**